code/modules/mob/living/simple_animal/hostile/mining_mobs/ice demon.dm

```diff
@@ -53,7 +53,9 @@
 	if(teleport_distance <= 0)
 		return ..()
 	var/list/possible_ends = list()
-	for(var/turf/T in view(teleport_distance, target.loc) - view(teleport_distance - 1, target.loc))
+	var/list/target_turfs = get_bounding_turfs(target.loc, teleport_distance, teleport_distance)
```

Comment:
the working theory for me replacing this is that this view call has to do extra work in some cases because there are a lot of objects around the target, plus we are discarding a bunch of atoms because we only operate on the turfs

Reply:
I also ripped it out of the for loop out of an abundance of caution for byond being boneheaded

Reply:
If this is more performant, should probably replace these as well.
https://github.com/tgstation/tgstation/blob/1aa293ea337283a0191140a878eeba319221e5df/code/__HELPERS/game.dm#L162

https://github.com/tgstation/tgstation/blob/1aa293ea337283a0191140a878eeba319221e5df/code/datums/diseases/advance/symptoms/heal.dm#L77-L79

https://github.com/tgstation/tgstation/blob/6ef421be4282bb15a70f09b4964bb517b9c6e26f/code/game/objects/items/grenades/antigravity.dm#L14

https://github.com/tgstation/tgstation/blob/1aa293ea337283a0191140a878eeba319221e5df/code/game/objects/items/grenades/syndieminibomb.dm#L61-L67

https://github.com/tgstation/tgstation/blob/131de978ca2d89f948cf1a6de7265b284c97b757/code/modules/mob/living/simple_animal/hostile/megafauna/wendigo.dm#L166-L172

https://github.com/tgstation/tgstation/blob/f1b4c5a289f74eef601499abb23a345603a7bceb/code/modules/flufftext/Hallucination.dm#L1058-L1060

https://github.com/tgstation/tgstation/blob/f1b4c5a289f74eef601499abb23a345603a7bceb/code/modules/flufftext/Hallucination.dm#L1270-L1272

https://github.com/tgstation/tgstation/blob/f1b4c5a289f74eef601499abb23a345603a7bceb/code/modules/flufftext/Hallucination.dm#L1300-L1303

https://github.com/tgstation/tgstation/blob/d0bb54087d2c97864a4922c1c8adc3fb1973a018/code/game/objects/items/devices/spyglasses.dm#L88-L89

https://github.com/tgstation/tgstation/blob/12b37e2b80d81c1ee9e825c79c1b924ca67008d7/code/game/objects/effects/glowshroom.dm#L115-L122

Reply:
it's not proven that this causes anything fyi, it's just an abundance of caution hedging on this.

I expect if it was an issue we would see it else where, but none of them appear to also do the list intersections